### DISCUSSION TITLE
Intergrating applications using reverse tunnels

### DISCUSSION THREAD

**Discussion by DasPhero** [gravitational/teleport#10954] [Q&A] [answered]
Hey there,

evaluating teleport for our enterprise network raised some questions concerning connecting exisiting applications.

Some of our developers are using GUI tools like HeidiSQL to manage mysql databases or applications/scripts are using ssh/http apis/...

How can you intergrate then into the teleport workflow?
E.g. creating a port forwarding for port 3306 to the db server to use with the GUI application might work, but doesn't this defeat the purpose of using a tool like teleport as this would enable a "malicious" user to just open a port forwarding to port 22 and use ssh like usual which we are trying to deny.
The same goes for http APIs of servers in our private network.

**Comment by scottpgallagher** [comment]
As long as the DB GUI supports certificates in it's settings users will login to teleport with `tsh login` and `tsh db login` and then point the DB GUI at those certificates. Our guides can walk you through that configuration: https://goteleport.com/docs/database-access/guides/gui-clients/

**Reply by DasPhero** [reply]
Oh boy.
Looks like I completely missed this part of the documentation.
Thank you.

But this is really specialized for databases.
Is there a way to connect to other ports like an http API of an internal Server, using the java http library for example, without utilizing an ssh tunnel?
I know that you can access http pages using the teleport client. But for this the user has to authenticate to the auth server. So I'm thinking there might be a programmatic way to utilize the proxy for connections.

**Comment by scottpgallagher** [comment, marked as answer]
If you looking to run commands against the API of an app running you can do that from the CLI (again after tsh login) similar to our docs on Grafana here: https://goteleport.com/docs/application-access/guides/api-access/
